### PR TITLE
A few post sheet improvements

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR improves the post bottom sheet in a few ways (followup to #1199).
* Remove ability to block oneself.
* Immediately hide posts from blocked communities/users. ([#298](https://github.com/thunder-app/thunder/issues/298))
* Ensure that Subscribe/Unsubscribe accurately reflects the current state.

### Demo - Block

https://github.com/thunder-app/thunder/assets/7417301/4d41a737-b352-484b-a51f-89d3126c53fd

### CODE REVIEW

**Comment by hjiangsu** [comment]
Thanks for working on the dismiss for blocked communities and users, it looks great!

I just have one question:

> Ensure that Subscribe/Unsubscribe accurately reflects the current state.

Does this only apply to the current post? (e.g., If I have multiple posts from a given community in the feed, but I subscribe to a community via a post card action, does the subscribed state apply to all other posts with that community?)

If not, then maybe a better way to handle this is to display a snackbar message when the subscribed action is successfully performed. I think the user will mostly understand that they have to refresh the feed to see that subscribed status.

**Comment by micahmo** [comment]
> Does this only apply to the current post?
> 
> If not, then maybe a better way to handle this is to display a snackbar message when the subscribed action is successfully performed.

Ahh, good catch, it did only apply to the given post. I replaced the update with a snackbar!

**Review by hjiangsu — APPROVED** [review]
Just a few minor comments!

File: lib/community/bloc/community_event.dart [hunk 1]
```diff
@@ -18,7 +18,10 @@ final class CommunityActionEvent extends CommunityEvent {
   /// TODO: Change the dynamic type to the correct type(s) if possible
   final dynamic value;
 
-  const CommunityActionEvent({required this.communityId, required this.communityAction, this.value});
+  /// An optional post associated with this action
+  final PostViewMedia? postViewMedia;
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/community/bloc/community_event.dart]: Good point, removed!

File: lib/community/bloc/community_bloc.dart [hunk 2]
```diff
@@ -6,8 +6,10 @@ import 'package:stream_transform/stream_transform.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 
 import 'package:thunder/community/enums/community_action.dart';
+import 'package:thunder/core/models/post_view_media.dart';
```
[the remark this answers is not in the document]
Comment (micahmo) [reply lib/community/bloc/community_bloc.dart]: Nope, removed!

**Pull request merged**